src/main/java/bookstore/security/constraint/FieldMatchValidator.java

```diff
@@ -0,0 +1,29 @@
+package bookstore.security.constraint;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import org.springframework.beans.BeanWrapperImpl;
+
+public class FieldMatchValidator implements ConstraintValidator<FieldMatch, Object> {
+    private String field;
+    private String fieldMatch;
+
+    @Override
+    public void initialize(FieldMatch constraintAnnotation) {
+        this.field = constraintAnnotation.field();
+        this.fieldMatch = constraintAnnotation.fieldMatch();
+    }
+
+    @Override
+    public boolean isValid(Object value, ConstraintValidatorContext context) {
+        Object fieldValue = new BeanWrapperImpl(value)
+                .getPropertyValue(field);
+        Object fieldMatchValue = new BeanWrapperImpl(value)
+                .getPropertyValue(fieldMatch);
+        if (fieldValue != null) {
```

Comment:
You can simplify your code
return fieldValue != null && fieldValue.equals(fieldMatchValue);